app/decorators/service_provider_session_decorator.rb

```diff
@@ -42,8 +54,8 @@ def sp_name
   end
 
   def sp_return_url
-    if request_url.present?
-      OpenidConnectRedirector.from_request_url(request_url).decline_redirect_uri
+    if sp.redirect_uri.present? && openid_connect_redirector.valid?
```

Comment:
Yeah it might be easiest just to leave it, I have some ideas about rescues but that would involve more changes and it might just be easier to stop the issues earlier here